Collections/FISAppDelegate.m

```diff
@@ -10,40 +10,193 @@
-    // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
+-(NSArray *)arrayByFilteringStringsBeginningWithA:(NSArray *)array {
+    
+    NSPredicate *beginsWithAPredicate = [NSPredicate predicateWithFormat:@"self BEGINSWITH[c] 'a'"];
+    NSArray *beginsWithA = [array filteredArrayUsingPredicate:beginsWithAPredicate];
+    
+    return beginsWithA;
 }
 
-- (void)applicationDidBecomeActive:(UIApplication *)application
-{
-    // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+
+-(NSInteger)sumOfIntegerNumbersInArray:(NSArray *)array {
+    
+    NSInteger sum = 0;
+    for (NSNumber *number in array) {
+        sum = sum + [number integerValue];
+    }
+    
+    return sum;
 }
 
-- (void)applicationWillTerminate:(UIApplication *)application
-{
-    // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+
+-(NSArray *)arrayByPluralizingStringsInArray:(NSArray *)array {
+    
+    NSMutableArray *plurals = [[NSMutableArray alloc] init];
+    
+    for (NSString *singular in array) {
+        NSString *plural = @"";
+        if ([singular containsString:@"oo"]) {
+            plural = [singular stringByReplacingOccurrencesOfString:@"oo" withString:@"ee"];
+        } else if ([singular containsString:@"ox"]) {
+            if ([singular hasPrefix:@"b"]) {
+                plural = [singular stringByAppendingString:@"es"];
+            } else {
+                plural = [singular stringByAppendingString:@"en"];
+            }
+        } else if ([singular hasSuffix:@"us"]) {
+            plural = [singular stringByReplacingOccurrencesOfString:@"us" withString:@"i"];
+        } else if ([singular hasSuffix:@"um"]) {
+            plural = [singular stringByReplacingOccurrencesOfString:@"um" withString:@"a"];
+        } else {
+            plural = [singular stringByAppendingString:@"s"];
+        }
+        [plurals addObject:plural];
+    }
+    
+    return [NSArray arrayWithArray:plurals];
 }
 
+-(NSDictionary *)countsOfWordsInString:(NSString *)string {
+    NSArray *punctuationStrings = @[@".", @",", @";", @"-"];
+    
+    NSString *noPunctuation = string;
+    
+    for (NSString *punctuation in punctuationStrings) {
+        noPunctuation = [noPunctuation stringByReplacingOccurrencesOfString:punctuation withString:@""];
+    }
+    
+    NSString *lowercaseNoPunctuation = [noPunctuation lowercaseString];
+    
+    NSArray *arrayOfWords = [lowercaseNoPunctuation componentsSeparatedByString:@" "];
+    
+    NSMutableDictionary *countsOfWords = [[NSMutableDictionary alloc] init];
+    
+    for (NSString *word in arrayOfWords) {
+        if ([countsOfWords[word] integerValue] > 0) {
+            NSInteger count = [countsOfWords[word] integerValue];
```

Comment:
haha, i was actually trying to improve readability by doing that. oh well, i'll do it on one line.